### ISSUE TITLE
Function parameters 'not accessed' and 'locals()' 

### ISSUE THREAD

**Issue by feluxe** [microsoft/pyright#7413]
**Is your feature request related to a problem? Please describe.**

We have a lot of wrapper functions in our code base using `locals()` like this:

```python
def foo(a=1, b=2, c=3, ...):
     print(a, b)
     
def public_foo_wrapped(a=1, b=2, c=3, ...):
    kwargs = locals()
    kwargs['a'] = do_stuff()
    foo(**kwargs)
```
Unfortunately this produces a "not accessed" warning for each parameter, which cannot be muted easily.

**Describe the solution you’d like**

Something like a `reportUnusedParameter` setting would be great!

**Comment by erictraut** [comment]
Pyright never emits a warning for not-accessed parameters. At most, it emits a tagged hint that tells a client to display the parameter with a grayed-out text effect.

Some clients don't properly support this tagged hint. If you're using such a client, you can disable the generation of this tagged hint by setting the `pyright.disableTaggedHints` [LSP setting](https://microsoft.github.io/pyright/#/settings?id=pyright-settings) to true.

**Comment by feluxe** [comment]
@erictraut 

> Pyright never emits a warning for not-accessed parameters. At most, it emits a tagged hint that tells a client to display the parameter with a grayed-out text effect.

Yes, it doesn't produce *warnings* but *hints*, I got that wrong, sorry. It's still annoying that these specific hints cannot be disabled, because the *issue views* get heavily polluted by them. Just an example from emacs:

![image](https://github.com/microsoft/pyright/assets/6545589/a0a0c0a2-317d-49d5-bfb7-2e2263e5968b)

> If you're using such a client, you can disable the generation of this tagged hint by setting the pyright.disableTaggedHints [LSP setting](https://microsoft.github.io/pyright/#/settings?id=pyright-settings) to true.

We don't want to disable them completely, just for the false positives when used with `locals()`.

We now have to live with hundreds of *hints* or disabled hints entirely. Both solutions are quite unfortunate.

**Comment by erictraut** [comment]
This isn't a problem for users of VS Code or other clients that display these tagged hints correctly. You might want to consider switching to a different editor if you find this annoying.
